src/overlays/actors/ovl_En_Pr2/z_en_pr2.c

```diff
@@ -71,48 +92,677 @@ const ActorInit En_Pr2_InitVars = {
+        this->actor.shape.rot.z = this->actor.world.rot.z;
+        this->unk_1D8 = 30;
+        this->actor.speedXZ = Rand_ZeroFloat(0.5f);
+        this->actor.world.rot.y = randPlusMinusPoint5Scaled(0x8000);
+    }
+    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 10);
+    this->unk_1D4 = 3;
+    this->actionFunc = func_80A75310;
+}
+
+void func_80A75310(EnPr2* this, GlobalContext* globalCtx) {
+    s32 temp;
+    f32 frame;
+    WaterBox* sp74;
+
+    SkelAnime_Update(&this->skelAnime);
+    if (this->unk_1E0 < 10) {
+        s32 i;
+
+        frame = this->skelAnime.curFrame;
+
+        if (this->unk_1F8 <= frame) {
+            for (i = 0; i < ARRAY_COUNT(this->unk_234); i++) {
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PR2, this->unk_234[i].x, this->unk_234[i].y,
+                            this->unk_234[i].z, this->actor.world.rot.x, this->actor.world.rot.y,
+                            this->actor.world.rot.z, i + 10);
+            }
+
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+    } else {
+        Vec3f sp64;
+
+        temp = false;
+        Math_SmoothStepToS(&this->actor.shape.rot.x, 0, 5, 10000, 1000);
+        Math_SmoothStepToS(&this->actor.shape.rot.z, this->unk_1E4, 5, 10000, 1000);
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->unk_1E6, 5, 10000, 1000);
+
+        if ((Rand_ZeroOne() < 0.3f) && (this->unk_1D6 == 0)) {
+            this->unk_1D6 = 1;
+        }
+
+        if (WaterBox_GetSurface1(globalCtx, &globalCtx->colCtx, this->actor.world.pos.x, this->actor.world.pos.z,
+                                 &this->unk_200, &sp74)) {
+            frame = this->unk_200 - 15.0f;
+
+            if (frame <= this->actor.world.pos.y) {
+                temp = true;
+            } else {
+                Math_ApproachF(&this->actor.world.pos.y, frame, 0.3f, 1.0f);
+            }
+        }
+
+        if ((this->unk_1D8 == 0) || temp) {
+            s32 i;
+
+            Math_SmoothStepToS(&this->unk_1F4, 0, 1, 15, 50);
+
+            if (this->unk_1F4 < 2) {
+                for (i = 0; i < 10; i++) {
+                    Math_Vec3f_Copy(&sp64, &this->actor.world.pos);
+
+                    sp64.x += randPlusMinusPoint5Scaled(20.0f);
+                    sp64.y += randPlusMinusPoint5Scaled(5.0f);
+                    sp64.z += randPlusMinusPoint5Scaled(20.0f);
+                    frame = Rand_ZeroFloat(0.03f) + 0.07f;
+
+                    EffectSsBubble_Spawn(globalCtx, &sp64, 0.0f, 5.0f, 5.0f, frame);
+                }
+
+                Actor_MarkForDeath(&this->actor);
+            }
+        }
+    }
+}
+
+void func_80A755D8(EnPr2* this, GlobalContext* globalCtx) {
+    static s16 D_80A75C3C[] = {
+        0x10 * 0, 0x10 * 1, 0x10 * 2,  0x10 * 3,  0x10 * 4,  0x10 * 5,  0x10 * 6,  0x10 * 7,
+        0x10 * 8, 0x10 * 9, 0x10 * 10, 0x10 * 11, 0x10 * 12, 0x10 * 13, 0x10 * 14, 0x10 * 15,
+    };
```

Comment:
What's the motivation for breaking this up instead of just using the second digit, i.e. `0x00, 0x10, 0x20, 0x30`, and so on?